snapcraft/cli.py

```diff
@@ -300,6 +300,9 @@ def run():  # noqa: C901 (complex-structure)
     except errors.LinterError as err:
         emit.error(craft_cli.errors.CraftError(f"linter error: {err}"))
         retcode = err.exit_code
+    except RemoteBuildFailedError as err:
```

Comment:
I don't think we should catch `RemoteBuild` and a subclassed error of `RemoteBuild`.  I say we keep it simple and only catch `RemoteBuild` errors.

Reply:
OK, the only downside is the error message become: `remote-build error: Remote build failed.`. It feels a little duplicative.